imnodes.cpp

```diff
@@ -2550,77 +2528,104 @@ void PopColorStyle()
     GImNodes->ColorModifierStack.pop_back();
 }
 
-float& LookupStyleVar(const ImNodesStyleVar item)
+struct ImNodesStyleVarInfo
 {
-    // TODO: once the switch gets too big and unwieldy to work with, we could do
-    // a byte-offset lookup into the Style struct, using the StyleVar as an
-    // index. This is how ImGui does it.
-    float* style_var = 0;
-    switch (item)
-    {
-    case ImNodesStyleVar_GridSpacing:
-        style_var = &GImNodes->Style.GridSpacing;
-        break;
-    case ImNodesStyleVar_NodeCornerRounding:
-        style_var = &GImNodes->Style.NodeCornerRounding;
-        break;
-    case ImNodesStyleVar_NodePaddingHorizontal:
-        style_var = &GImNodes->Style.NodePaddingHorizontal;
-        break;
-    case ImNodesStyleVar_NodePaddingVertical:
-        style_var = &GImNodes->Style.NodePaddingVertical;
-        break;
-    case ImNodesStyleVar_NodeBorderThickness:
-        style_var = &GImNodes->Style.NodeBorderThickness;
-        break;
-    case ImNodesStyleVar_LinkThickness:
-        style_var = &GImNodes->Style.LinkThickness;
-        break;
-    case ImNodesStyleVar_LinkLineSegmentsPerLength:
-        style_var = &GImNodes->Style.LinkLineSegmentsPerLength;
-        break;
-    case ImNodesStyleVar_LinkHoverDistance:
-        style_var = &GImNodes->Style.LinkHoverDistance;
-        break;
-    case ImNodesStyleVar_PinCircleRadius:
-        style_var = &GImNodes->Style.PinCircleRadius;
-        break;
-    case ImNodesStyleVar_PinQuadSideLength:
-        style_var = &GImNodes->Style.PinQuadSideLength;
-        break;
-    case ImNodesStyleVar_PinTriangleSideLength:
-        style_var = &GImNodes->Style.PinTriangleSideLength;
-        break;
-    case ImNodesStyleVar_PinLineThickness:
-        style_var = &GImNodes->Style.PinLineThickness;
-        break;
-    case ImNodesStyleVar_PinHoverRadius:
-        style_var = &GImNodes->Style.PinHoverRadius;
-        break;
-    case ImNodesStyleVar_PinOffset:
-        style_var = &GImNodes->Style.PinOffset;
-        break;
-    default:
-        assert(!"Invalid StyleVar value!");
-    }
+    ImGuiDataType   Type;
+    ImU32           Count;
+    ImU32           Offset;
+    void*           GetVarPtr(ImNodesStyle* style) const { return (void*)((unsigned char*)style + Offset); }
+};
+
+static const ImNodesStyleVarInfo GStyleVarInfo[] =
+{
+    // ImNodesStyleVar_GridSpacing
+    { ImGuiDataType_Float, 1, (ImU32)IM_OFFSETOF(ImNodesStyle, GridSpacing) },
+    // ImNodesStyleVar_NodeCornerRounding
+    { ImGuiDataType_Float, 1, (ImU32)IM_OFFSETOF(ImNodesStyle, NodeCornerRounding) },
+    // ImNodesStyleVar_NodePaddingHorizontal
+    { ImGuiDataType_Float, 1, (ImU32)IM_OFFSETOF(ImNodesStyle, NodePaddingHorizontal) },
+    // ImNodesStyleVar_NodePaddingVertical
+    { ImGuiDataType_Float, 1, (ImU32)IM_OFFSETOF(ImNodesStyle, NodePaddingVertical) },
```

Comment:
Now that there's a mechanism for adding `ImVec2` style variables, what do you think about changing these as well? As it is now, the style variables are a bit inconsistent 🙂

Reply:
Sure, refactored it.